runtime/memprof.c

```diff
@@ -1499,7 +1499,7 @@ static bool entry_update_after_minor_gc(entry_t e, void *data)
   CAMLassert(Is_block(e->block)
              || e->deleted || e->deallocated || e->offset);
   if (!e->offset && Is_block(e->block) && Is_young(e->block)) {
-    if (Hd_val(e->block) == 0) {
+    if (Is_promoted_hd(caml_get_header_val(e->block))) {
```

Comment:
This is a nontrivial change, which is shaped like a bugfix. Is there actually a bug here to fix? (I thought this function ran after the minor GC barrier when `caml_get_header_val` was no longer necessary)